### ISSUE TITLE
'pwd' command emits garbage at 2nd subdirectory level

### ISSUE THREAD

**Issue by dlkeng** [sparkfun/OpenLog#133]
**OpenLog v3.13**

The "pwd" command emits garbage at 2nd subdirectory level.
For example, with the following directory structure:

```
>ls
Volume is FAT32
CONFIG.TXT     48
ZZ1/
  ZZ2/
```

For example:

```
 0) board reset
 1) 'pwd' shows ".\"
 2) 'cd ZZ1'
 3) 'pwd' shows ".\ZZ1"
 4) 'cd ZZ2'
 5) 'pwd' shows ".\ %@#$.!($*&!@(#$*&(!@*&$!@($*&."  (and lots more!)
```

This appears to be caused by the following lines in the code:

```
1575      tmp_var = getNextFolderTreeIndex();
1576      for (byte i = 0; i < tmp_var; i++)
```

In that code, ' **tmp_var** ' and ' **i** ' are both unsigned, but _getNextFolderTreeIndex()_ returns -1 when limited by **FOLDER_TRACK_DEPTH**. So, in essence, the for() loop runs from 0 to 254 when a -1 is returned by  _getNextFolderTreeIndex()_ when the **folderTree[]** array is full.This results in trying to print file/directory names from data well off the end of the **folderTree[]** array.

Additionally, the whole operations traversing the directory tree and showing its current location behaves strangely.

The following is an example directory tree transversal operations and results

```
 1) 'pwd' shows ".\"
 2) 'cd ZZ1'
 3) 'pwd' shows ".\ZZ1"
 4) 'cd ..'
 5) 'pwd' shows ".\"
 6) 'cd ZZ1'
 7) 'pwd' shows ".\ZZ1"
 8) 'cd ZZ2'                 <-------- folderTree[] is now full
 9) 'pwd' shows ".\" %@#$..  <-------- is incorrect since is actually at ".\ZZ1\ZZ2"
10) 'cd ..'                  <-------- actually goes to root
11) 'pwd' shows ".\"
12) 'cd ZZ1'
13) 'pwd' shows ".\"         <-------- is incorrect since is actually at ".\ZZ1"
```

In summary, once **folderTree[]** has been filled, all subsequent 'pwd' commands show
is at root directory and all 'cd ..' commands always go to root directory! It appears the intent originally was to somehow limit the depth of the directory structure, but the 'cd' command is not limited on its tranversal down the directory structure, only the upward transversal.
